### PR TITLE
fix issues with REST retry logic

### PR DESCRIPTION
**Pull request by payes**
kubectl-mayastor plugin doesn't return error when the get volumes/nodes call fails. 
Therefore we don't retry getting the volume/node status using other IPs.
Signed-off-by: Payes Anand <payes.anand@mayadata.io>

### CODE REVIEW

File: src/common/controlplane/v1/msn_cp.go [hunk 1]
```diff
@@ -41,7 +42,7 @@ func GetMayastorCpNode(nodeName string, address []string) (*MayastorCpNode, erro
 		url := fmt.Sprintf("http://%s:%s", addr, common.PluginPort)
 		cmd := exec.Command(pluginpath, "-r", url, "-ojson", "get", "node", nodeName)
 		jsonInput, err = cmd.CombinedOutput()
-		if err == nil {
+		if err == nil && !strings.Contains(string(jsonInput), "error in request: request timed out") {
 			break
 		} else {
 			logf.Log.Info("Error while executing kubectl mayastor command", "node IP", addr, "error", err)
```
Comment (chriswldenyer) [inline src/common/controlplane/v1/msn_cp.go:48]: err could now be nil here, if there was a timeout,

File: src/common/controlplane/v1/msn_cp.go [hunk 2]
```diff
@@ -41,7 +42,7 @@ func GetMayastorCpNode(nodeName string, address []string) (*MayastorCpNode, erro
 		url := fmt.Sprintf("http://%s:%s", addr, common.PluginPort)
 		cmd := exec.Command(pluginpath, "-r", url, "-ojson", "get", "node", nodeName)
 		jsonInput, err = cmd.CombinedOutput()
-		if err == nil {
+		if err == nil && !strings.Contains(string(jsonInput), "error in request: request timed out") {
```
Comment (chriswldenyer) [inline src/common/controlplane/v1/msn_cp.go:45]: Could this test be a function, or maybe have the matched string be a constant? The error message we are catching might change in the future.

**Review by chriswldenyer — APPROVED** [review]
LGTM, but some suggestions.

**Review by blaisedias — APPROVED** [review]
Does `IsTimeoutError` need changing in  `src/common/controlplane/v1/controlplane_v1.go`

**Comment by payes** [comment]
bors try

**Pull request merged**